source/slang/slang.cpp

```diff
@@ -785,13 +1036,15 @@ SlangResult FrontEndCompileRequest::executeActionsInner()
     if (getSink()->GetErrorCount() != 0)
         return SLANG_FAIL;
 
-    if ((compileFlags & SLANG_COMPILE_FLAG_NO_CODEGEN) == 0)
-    {
-        // Generate initial IR for all the translation
-        // units, if we are in a mode where IR is called for.
-        generateIR();
-    }
-
+    // We always generate IR for all the translation units.
```

Comment:
What happened to NO_CODEGEN? That I know people are using it to find dependencies.

Reply:
The flags in the compiler were a bit of a mess, because there were two similarly-named options that did two nearly-but-not-quite-the-same things:

1. There was an option to suppress final kernel code generation (equivalent to not having a target, really), but that *didn't* suppress IR code generation.

2. There was an option to suppress *all* code generation, including IR code generation.

I believe what I've done here is eliminate option (2) and kept option (1), so the main cost for somebody who just wants to resolve dependencies is that we end up generating IR from the type-checked AST, instead of skipping that step. I could try to restore the old behavior, but we'd need two distinct flags/options for this, because Falcor needs (1), and that is the most sane default for the average user.